src/main/kotlin/hs/flensburg/marlin/business/schedulerJobs/sensorData/boundary/ReverseGeoCodingService.kt

```diff
@@ -1,71 +1,96 @@
 
-                // fetch the location name from Nominatim
-                runBlocking {
-                    val response = httpclient.get("https://nominatim.openstreetmap.org/reverse") {
-                        parameter("lat", lat)
-                        parameter("lon", lon)
-                        parameter("format", "json")
-                        header("User-Agent", "SOOP-Application/1.0")
-                    }.body<NominatimResponse>()
+                if (location.name.isNullOrBlank()) {
+                    name = parts.firstOrNull().orEmpty()
 
-                    println(response)
+                    val cityOrTown = response.address?.city
+                        ?: response.address?.town
+                        ?: response.address?.village
+                        ?: response.address?.suburb
+                        ?: ""
 
-                    var shortName = "Unknown Location"
+                    // Append city name if not already part of the name
+                    if (!cityOrTown.isBlank()) {
+                        val normalizedName = name.lowercase()
+                        val normalizedCity = cityOrTown.lowercase()
 
-                    if (response.display_name != null) {
-                        // Extract only the name
-                        shortName = response.display_name.split(",")[0].trim()
+                        if (!normalizedName.contains(normalizedCity)) {
+                            name = "$name, $cityOrTown"
+                        }
                     }
-                    // Update the location name in the database
-                    update(LOCATION)
-                        .set(LOCATION.NAME, shortName)
-                        .where(LOCATION.ID.eq(id))
-                        .execute()
                 }
-            } catch (e: Exception) {
-                // report the error
-                println("Fehler beim Geocoding für Location $id: ${e.message}")
+
+                if (location.address.isNullOrBlank()) {
+                    address = parts.drop(1).joinToString(", ").ifBlank { address }
+                }
             }
+
+            !LocationRepo.updateLocation(
+                id = location.id!!,
+                name = name,
+                address = address,
+                description = location.description,
+                openingHours = location.openingHours,
+                phone = location.contactPhone,
+                email = location.contactEmail,
+                website = location.contactWebsite
+            ).orDie()
         }
+        KIO.unit
+    }
+
+    fun fetchLocationInfoFromNominatim(lat: Double,lon: Double): NominatimResponse = runBlocking {
+        httpclient.get("https://nominatim.openstreetmap.org/reverse") {
+            parameter("lat", lat)
+            parameter("lon", lon)
+            parameter("format", "json")
+            header("User-Agent", "SOOP-Application/1.0")
```

Comment:
Solche Header würde ich unter httpclient konfigurieren,
```suggestion
val httpclient = HttpClient(CIO) {
    headers {
        append("User-Agent", "Marlin-Live")
        // Hier kann man bei Bedarf weitere hinzufügen.
    }

    install(ContentNegotiation) {
        json(Json {
            ignoreUnknownKeys = true
        })
    }

    install(HttpTimeout) {
        requestTimeoutMillis = 30000
        connectTimeoutMillis = 30000
        socketTimeoutMillis = 30000
    }
}
```